doc/authentication.md

```diff
@@ -0,0 +1,73 @@
+Requiring authentication
+========================
+
+In some cases, it might be needed for security or legal reasons to secure the
+projects' code available through CodeCompass against unauthorised access.
+
+To enable this feature, an `authentication.json` file should be created under
+the workspace directory (`--workspace` or `-w` flag given to the server).
+
+At a bare minimum, to restrict access, an `"enabled": true` MUST be present
+in the JSON.
+
+An example valid JSON looks like this:
+
+~~~~{.json}
+{
+    "enabled": true,
+    "prompt": "CodeCompass tutorial server",
+    "session-lifetime": 14400,
```

Comment:
I'll change to use _s instead of -s to be similar to CodeChecker's equivalent file, if it's necessary.

This JSON is only accessed by the web server, never exposed to JavaScript. In the C++ code, you'll have to use `operator[]` either way.